spec/latest/index.bs

```diff
@@ -223,12 +225,12 @@ Data Types {#key_data_types}
 ----------------------------
 
 The `data_types` key must be present and shall define the data types of all required
-arrays based on the [[#key_format]]. The data type declares the type of the
-in-memory arrays. While these are often identical to the types used when storing
-the arrays on disk in the container, the container may choose to store the arrays
-in another format. For example, `uint64` type may be stored as `int8` if all the
-numbers in the array are small enough to fit, but `data_types` would still list the
-array as having type `uint64`.
+arrays based on the [[#key_format]]. The data type declares the type of both the
+on-disk array as well as the in-memory array. When these are identical, a simple string
+defines the type. When the on-disk and in-memory types differ due to limitations in the
+storage container (ex. HDF5 lacks a BOOL type), the type is shown as "on_disk->in_memory".
+For the example of storing BOOL type as INT8, this would be "int8->bool" to indicate that
+after reading the values array into memory, it should be interpreted as boolean data.
```

Comment:
I'd totally forgotten about this bit of hdf5.

* Do you think there are other cases where this kind of transformation is unambiguous?
* I'm wondering if we need to set a list of "known dtypes". Or is `uint8->foo` fine?